ratlib/starsystem.py

```diff
@@ -396,44 +396,44 @@ def refresh_bloom(bot, db=None):
     bot.memory['ratbot']['stats']['starsystem_bloom'] = {'entries': count, 'time': t.seconds}
     return bloom
 
-def sysapi_query(system, querytype):
-    if querytype == "search":
-        try:
-            response = requests.get('https://system.api.fuelrats.com/search?name={}'.format(system))
-            if response.status_code != 200:
-                return { "meta": { "error": "System API did not respond with valid data." } }
-            result = response.json()
-        except Timeout:
-            return { "meta": { "error": "The request to Systems API timed out!"} }
-        return result
+def sysapi_query(system, querytype=None):
+    """
+    Queries systems api for name matches or landmarks.
+    """
+
+    queryurl = 'https://system.api.fuelrats.com/'
+
     if querytype == "landmark":
-        try:
-            response = requests.get('https://system.api.fuelrats.com/landmark?name={}'.format(system))
-            if response.status_code != 200:
-                return { "meta": { "error": "System API did not respond with valid data."} }
-            result = response.json()
-        except Timeout:
-            return { "meta": { "error": "The request to Systems API timed out!"} }
-        return result
+        queryurl += 'landmark?name={}'
+    elif querytype == "smart":
+        queryurl += 'mecha?name={}'
     else:
-        try:
-            response = requests.get('https://system.api.fuelrats.com/api/systems?filter[name:eq]={}'.format(system))
-            if response.status_code != 200:
-                return { "meta": { "error": "System API did not respond with valid data."} }
-            result = response.json()['data']
-        except Timeout:
-            return { "meta": { "error": "The request to Systems API timed out!"} }
-        return result
+        queryurl += 'search?name={}'
+
+    try:
+        response = requests.get(queryurl.format(system))
+        if response.status_code != 200:
+            return { "meta": { "error": "System API did not respond with valid data." } }
```

Comment:
[question] wouldn't a non-2xx response be invalid?
shouldn't we actually /validate/ the returned object before making this assertion?

Reply:
not sure what you're asking here. Do you mean validate response?

Reply:
```py
return { "meta": { "error": "System API did not respond with valid data." } }
```